tests/data_test.py

```diff
@@ -1,47 +1,58 @@
 from pathlib import Path
+from typing import Union
 
 import numpy as np
 import pandas as pd
 import pytest
 import torch
 import torchaudio.transforms as AT
 from torchvision import transforms as T
-from torchvision.datasets import VisionDataset
 from typing_extensions import Type
 
 from conduit.data import (
+    NICO,
+    SSRP,
     BinarySample,
     BinarySampleIW,
+    CelebA,
     NamedSample,
     TernarySample,
     TernarySampleIW,
+    Waterbirds,
 )
 from conduit.data.datamodules import EcoacousticsDataModule
 from conduit.data.datamodules.tabular.dummy import DummyTabularDataModule
 from conduit.data.datasets import ISIC, ColoredMNIST, Ecoacoustics
 
 
 @pytest.mark.slow
-@pytest.mark.parametrize("ds_cls", [ColoredMNIST, ISIC])
-def test_datasets(ds_cls: Type[VisionDataset]) -> None:
+@pytest.mark.parametrize("ds_cls", [ColoredMNIST, ISIC, CelebA, NICO, SSRP, Waterbirds])
+def test_vision_datasets(
+    root: Path,
+    ds_cls: Union[
+        Type[ColoredMNIST], Type[ISIC], Type[CelebA], Type[NICO], Type[SSRP], Type[Waterbirds]
+    ],
+) -> None:
     """Basic test for datasets.
     Confirms that the datasets can be instantiated and have a functional __getitem__ method.
     """
     transform = T.ToTensor()
-    ds = ds_cls(root="~/Data", transform=transform)
+    ds = ds_cls(root=root, transform=transform)
     for _ds in ds:
-        assert _ds[0] is not None
+        assert isinstance(_ds, TernarySample)
+        assert _ds.x[0] is not None
 
 
 @pytest.mark.slow
-def test_ecoacoustics_dataset() -> None:
-    root_dir = Path("~/Data").expanduser()
+@pytest.mark.parametrize("ds_cls", [Ecoacoustics])
+def test_audio_dataset(root: Path, ds_cls: Type[Ecoacoustics]) -> None:
+    root_dir = Path(root).expanduser()
     base_dir = root_dir / "Ecoacoustics"
     target_attribute = "habitat"
```

Comment:
If you're parameterizing this test, then this should probably be parameterized as well

Reply:
yeah, the problem, or rather, what I assume is a problem, that I don't want to touch, is that the "preprocessing" doesn't save in specific dirs